my-code/src/containers/Home.js

```diff
@@ -0,0 +1,105 @@
+import React, { Component } from 'react'
+import SearchInput from '../components/searchInput';
+import MovieCard from '../components/movieCard';
+import logo from '../assets/images/logo.svg';
+import { withRouter } from 'react-router-dom';
+import EmptyStateManager from '../components/emptyStateManager';
+
+class Home extends Component {
+
+  constructor(props) {
+    super(props)
+    this.state = {
+      data: [],
+      searchValue: '',
+      loading: false,
+      error: '',
+      addedMovies: [],
+    }
+  }
+
+  componentDidMount() {
+    if(localStorage.getItem('addedMovies')) {
+      const retrievedData = localStorage.getItem('addedMovies');
+      const parseData = JSON.parse(retrievedData);
+      this.setState({ addedMovies: parseData})
+    }
+  }
+
+  handleChange(event) {
+    const value = event.target.value;
+    this.setState({ searchValue: value,loading: true, error: '' },
+      () => {
+        this.fetchMovies(value);
+      }
+    );
+  }
+
+  fetchMovies(value) {
+    fetch(`http://www.omdbapi.com/?apikey=a6487992&s=${value}` , {
+      method: 'get',
+      headers: {
+        'Accept': 'application/json',
+      },
+    })
+    .then((response) => { return response.json(); })
+    .then((response) => {
+      if (response.Response === 'True') {
+        this.setState({ data: response.Search, loading: false, error: '' })
+      } else {
+        this.setState({ data: [], loading: false, error: response.Error })
+      }
+    })
+    .catch((error) => {
+      console.log(error)
+    })
+  }
+
+  handleAddButton = (id) => {
+    console.log(this.state)
+    if(this.state.addedMovies.includes(id)) {
```

Comment:
I would recommend splitting this function into small pieces, there are too many things happening here. For example, the local storage handle might be a util.